src/plugins/dashboard/public/application/embeddable/viewport/dashboard_viewport.tsx

```diff
@@ -152,7 +176,7 @@ export class DashboardViewport extends React.Component<DashboardViewportProps, S
   public render() {
     return (
       <React.Fragment>
-        {this.state.isEmptyState ? this.renderEmptyScreen() : null}
+        {this.shouldRenderEmpty() ? this.renderEmptyScreen() : null}
```

Comment:
Methods that return React components to render are a code smell that indicates the method should actually be a stand alone component, (and one that receives props sufficient for it to determine if it should render null).

Reply:
Changed up the dashboard_viewport component to hopefully make it easier to grok.